scapy/main.py

```diff
@@ -317,7 +317,7 @@ def save_session(fname="", session=None, pickleProto=-1):
     if not fname:
         fname = conf.session
         if not fname:
-            conf.session = fname = utils.get_temp_file(keep=True)
+            conf.session = fname = cast(str, utils.get_temp_file(keep=True))
```

Comment:
Why do we need to cast to `str` here?

Reply:
`get_temp_file` can return a file descriptor if `fd=True`

Reply:
Thanks. I missed that.